app/src/views/features/rules/RuleEditor/RuleEditor.js

```diff
@@ -44,70 +76,41 @@ const RuleEditor = () => {
-        case GLOBAL_CONSTANTS.RULE_TYPES.HEADERS:
-        case GLOBAL_CONSTANTS.RULE_TYPES.REQUEST:
-        case GLOBAL_CONSTANTS.RULE_TYPES.RESPONSE:
-        case GLOBAL_CONSTANTS.RULE_TYPES.DELAY:
-        case GLOBAL_CONSTANTS.RULE_TYPES.USERAGENT:
-        case GLOBAL_CONSTANTS.RULE_TYPES.SCRIPT:
-          return (
-            <>
-              <Split
-                sizes={rulePaneSizes}
-                minSize={0}
-                gutterSize={20}
-                dragInterval={20}
-                direction="vertical"
-                cursor="row-resize"
-                onDragEnd={(paneSizes) => setRulePaneSizes(paneSizes)}
-                style={{
-                  height: location.pathname.includes("/rules/editor") ? "calc(100vh - 72px)" : "91vh",
-                }}
-              >
-                <Row className="overflow-hidden">
-                  <Col span={24}>
-                    <ProCard className="rule-editor-procard">
-                      <RuleBuilder />
-                    </ProCard>
-                  </Col>
-                </Row>
-                <Row className="overflow-hidden">
-                  <ProCard className="rule-editor-split-procard">
-                    <RuleEditorSplitPane
-                      mode={MODE}
-                      showExecutionLogs={true}
-                      expandRulePane={expandRulePane}
-                      collapseRulesPlane={collapseRulesPlane}
-                      ruleType={CURRENT_RULE_TYPE}
-                    />
-                  </ProCard>
-                </Row>
-              </Split>
-            </>
-          );
+  useEffect(() => {
+    if (isNewRuleCreated) {
+      toggleBottomSheet();
+      setIsNewRuleCreated(false);
+    }
+  }, [toggleBottomSheet, isNewRuleCreated]);
 
-        default:
-          break;
-      }
+  useEffect(() => {
+    if (state?.source === APP_CONSTANTS.RULE_EDITOR_CONFIG.MODES.CREATE) {
+      setIsNewRuleCreated(true);
     }
+  }, [state?.source, MODE]);
+
+  const renderRuleEditor = () => {
     return (
-      <>
-        <ProCard className="rule-editor-procard">
-          <RuleBuilder />
-        </ProCard>
-      </>
+      <Col className="overflow-hidden h-full">
+        {MODE !== RULE_EDITOR_CONFIG.MODES.SHARED_LIST_RULE_VIEW ? (
+          <EditorHeader
+            mode={MODE}
+            location={location}
+            currentlySelectedRuleData={currentlySelectedRuleData}
+            currentlySelectedRuleConfig={currentlySelectedRuleConfig}
+          />
+        ) : null}
+        <BottomSheetLayout
+          contentWidth={sheetPlacement === BottomSheetPlacement.BOTTOM ? 24 : 13}
+          sheetWidth={sheetPlacement === BottomSheetPlacement.BOTTOM ? 24 : 11}
+          isSheetHidden={!(MODE === RULE_EDITOR_CONFIG.MODES.EDIT && isFeatureCompatible(FEATURES.TEST_THIS_RULE))}
+          bottomSheet={<BottomSheet defaultActiveKey={BOTTOM_SHEET_TAB_KEYS.TEST_RULE} items={bottomSheetTabItems} />}
```

Comment:
We shouldn't pass `contentWidth`, `sheetWidth`, `isSheetHidden` for the sake of consistency. In future if we decide to make sidepanel width configurable, we can just pass it as prop and let the layout handle everything else

Reply:
You can abstract BottomSheet component in a separate component named `RuleEditorBottomSheet`